From 9d0f2fbdde60fe22c78d032c72c94c926f715371 Mon Sep 17 00:00:00 2001
From: PyScotch <pyscotch@noreply.invalid>
Date: Tue, 13 Oct 2026 20:50:41 +0000
Subject: [PATCH 232/240] Bugfix: release the mesh and its arrays in the
 Fortran mesh build test program [report PyScotch]

"test_scotch_mesh_build_f.f90" built a mesh but never called
scotchfmeshexit, and never deallocated the "verttab" and "edgetab"
arrays it allocated, unlike "test_libmetis_dual_f.f90". The program
therefore fails as soon as it is run under LeakSanitizer, which the
"check" target now does. Release the mesh and the arrays before the
end of the program.

Refs: PS2-G17-7
---
 src/check/test_scotch_mesh_build_f.f90.in | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/check/test_scotch_mesh_build_f.f90.in b/src/check/test_scotch_mesh_build_f.f90.in
index 3868b35..b8c1272 100644
--- a/src/check/test_scotch_mesh_build_f.f90.in
+++ b/src/check/test_scotch_mesh_build_f.f90.in
@@ -39,6 +39,7 @@
 !*                the mesh building routine.              **
 !*                                                        **
 !*   DATES      : # Version 7.0  : from : 24 jan 2026     **
+!*                                 to   : 13 oct 2026     **
 !*                                 to     13 oct 2026     **
 !*                                                        **
 !234567*****************************************************
@@ -99,4 +100,9 @@ program test_scotch_mesh_build
     call exit_c (1)
   end if
 
+  call scotchfmeshexit##SCOTCH_NAME_SUFFIX(meshdat)
+
+  deallocate (edgetab)
+  deallocate (verttab)
+
 end program test_scotch_mesh_build
-- 
2.47.3

